Homework4/Program.cs

```diff
@@ -4,7 +4,189 @@ namespace Homework4
+                array[i] = random.Next(0, 100);
+            }
+
+            for (int i = 0; i < size; i++)
+            {
+                Console.Write(array[i] + " ");
+            }
+
+            Console.WriteLine();
+
+            return array;
+        }
+
+        static int FindMinElement(int[] arr)
+        {
+            int minI = 0;
+
+            for (int i = 1; i < arr.Length; i++)
+            {
+                if (arr[i] < arr[minI])
+                {
+                    minI = i;
+                }
+            }
+
+            return arr[minI];
+        }
+
+        static int FindMaxElement(int[] arr)
+        {
+            int maxI = 0;
+
+            for (int i = 1; i < arr.Length; i++)
+            {
+                if (arr[i] > arr[maxI])
+                {
+                    maxI = i;
+                }
+            }
+
+            return arr[maxI];
+        }
+
+        static int FindMinIndex(int[] arr)
+        {
+            int minI = 0;
+
+            for (int i = 1; i < arr.Length; i++)
+            {
+                if (arr[i] < arr[minI])
+                {
+                    minI = i;
+                }
+            }
+
+            return minI;
+        }
+
+        static int FindMaxIndex(int[] arr)
+        {
+            int maxI = 0;
+
+            for (int i = 1; i < arr.Length; i++)
+            {
+                if (arr[i] > arr[maxI])
+                {
+                    maxI = i;
+                }
+            }
+
+            return maxI;
+        }
+
+        static int CalcSumOddElements(int[] arr)
+        {
+            int sum = 0;
+
+            for (int i = 1; i < arr.Length; i++)
+            {
+                if (i % 2 == 1)
+                {
+                    sum += arr[i];
+                }
+            }
+
+            return sum;
+        }
+
+        static void Swap(ref int a, ref int b)
+        {
+            int temp = a;
+            a = b;
+            b = temp;
+        }
+
+        static int[] ReverseArray(int[] arr)
+        {
+            int lastIndex = arr.Length - 1;
+
+            for (int i = 0; i < (arr.Length / 2); i++)
+            {
+                Swap(ref arr[i], ref arr[lastIndex]);
+                lastIndex--;
+            }
+
+            return arr;
+        }
+
+        static int CountOddElements(int[] arr)
+        {
+            int count = 0;
+
+            for (int i = 0; i < arr.Length; i++)
+            {
+                if (arr[i] % 2 != 0)
+                {
+                    count += arr[i];
+                }
+            }
+
+            return count;
+        }
+
+        static int[] SwapHalfs(int[] arr)
+        {
+            int middle = arr.Length / 2;
+            int startLastHalf = arr.Length - middle;
+
+            for (int i = 0; i < middle; i++)
+            {
+                Swap(ref arr[i], ref arr[startLastHalf]);
+                startLastHalf++;
+            }
+
+            return arr;
+        }
```

Comment:
What about, if array will have odd indexes? Like in task "..or for 1 2 3 4 5 -> 4 5 3 1 2"